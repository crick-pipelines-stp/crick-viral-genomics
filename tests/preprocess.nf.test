nextflow_pipeline {
    name "Test Pipeline Components"
    script "../main.nf"

    tag "preprocess"

    test("preprocess_cat_fastq") {
        tag "preprocess/cat_fastq"
        when {
            params {
                run_reporting = false

                outdir      = "$outputDir"
                viral_fasta = "$projectDir/data/refs/covid/SARS-CoV-2.fasta"
                samplesheet = "$projectDir/tests/data/samplesheets/test_nanopore_cov_amp.csv"
            }
        }
        then {
            def stable_name = getAllFilesFromDir(params.outdir, relative: true, includeDir: true, ignore: ['pipeline_info/*.{html,json,txt}'])
            def stable_path = getAllFilesFromDir(params.outdir, ignoreFile: 'tests/.nftignore')
            def task_count = workflow.trace.tasks().count { task ->
                task.name.contains("CAT_FASTQ")
            }
            assertAll(
                { assert workflow.success},
                { assert task_count == 2 },
                { assert snapshot(
                    workflow.trace.succeeded().size(),
                    stable_name,
                    stable_path
                ).match() }
            )
        }
    }

    test("preprocess_trim_illumina") {
        tag "preprocess/trim_illumina"
        when {
            params {
                run_illumina_qc_trim = true
                run_reporting        = false

                outdir      = "$outputDir"
                viral_fasta = "$projectDir/tests/data/seq_sim/influenza_A/*"
                samplesheet = "$projectDir/tests/data/samplesheets/test_illumina_flu_amp_small.csv"
            }
        }
        then {
            def stable_name = getAllFilesFromDir(params.outdir, relative: true, includeDir: true, ignore: ['pipeline_info/*.{html,json,txt}'])
            def stable_path = getAllFilesFromDir(params.outdir, ignoreFile: 'tests/.nftignore')
            assertAll(
                { assert workflow.success},
                { assert snapshot(
                    workflow.trace.succeeded().size(),
                    stable_name,
                    stable_path
                ).match() }
            )
        }
    }

    test("preprocess_trim_nanopore") {
        tag "preprocess/trim_nanopore"
        when {
            params {
                run_nanopore_qc_trim = true
                run_reporting        = false

                outdir      = "$outputDir"
                viral_fasta = "$projectDir/data/refs/covid/SARS-CoV-2.fasta"
                samplesheet = "$projectDir/tests/data/samplesheets/test_nanopore_cov_amp.csv"
            }
        }
        then {
            def stable_name = getAllFilesFromDir(params.outdir, relative: true, includeDir: true, ignore: ['pipeline_info/*.{html,json,txt}'])
            def stable_path = getAllFilesFromDir(params.outdir, ignoreFile: 'tests/.nftignore')
            assertAll(
                { assert workflow.success},
                { assert snapshot(
                    workflow.trace.succeeded().size(),
                    stable_name,
                    stable_path
                ).match() }
            )
        }
    }
}
