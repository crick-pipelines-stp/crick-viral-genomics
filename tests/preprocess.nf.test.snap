{
    "preprocess_cat_fastq": {
        "content": [
            6,
            [
                "consensus",
                "consensus/consensus.merged.fasta",
                "pipeline_info",
                "ref_assembly",
                "ref_assembly/SARS-CoV-2.fasta.fai"
            ],
            [
                "consensus.merged.fasta:md5,d41d8cd98f00b204e9800998ecf8427e",
                "SARS-CoV-2.fasta.fai:md5,6806a14930eab7e2de4e9776d79bcab2"
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.4"
        },
        "timestamp": "2025-02-07T10:34:39.68796"
    }
}